code/game/gamemodes/objective.dm

```diff
@@ -872,3 +828,21 @@ GLOBAL_LIST_INIT(potential_theft_objectives, (subtypesof(/datum/theft_objective)
 	explanation_text = "Hunger grows within us, we need to feast on the brains of the uninfected. Scratch, bite, and spread the plague."
 	needs_target = FALSE
 	completed = TRUE
+
+// Placeholder objectives that will replace themselves
+
+/datum/objective/delayed
+	needs_target = FALSE
+	var/objective_to_replace_with
+
+/datum/objective/delayed/New(datum/objective/delayed_objective)
+	objective_to_replace_with = delayed_objective
```

Comment:
Here you can make it randomly create an objective instead if it is null. Or it should ensure/check that the parameter is not null.

Reply:
imo it should have a stack_trace if theres no `objective_to_replace_with` set in both /New() and the reveal proc

Reply:
A stack trace is better imo